File: fixtures/adapter-swift/basic/UserService.swift
// Everything below is resolvable inside this one file, which is exactly the
// limit of Tier 2: the analyzer records what it can see here and nothing more.

protocol Auditing {
    func describe() -> String
}

struct AuditLog {
    func record(_ message: String) -> String {
        return message
    }
}

final class UserService: Auditing {
    private let log = AuditLog()

    func describe() -> String {
        return "user service"
    }

    func handle() -> String {
        // A call to a method defined in this file.
        return describe()
    }
}

File: fixtures/adapter-php/basic/UserService.php
<?php
// Everything below is resolvable inside this one file, which is exactly the
// limit of Tier 2: the analyzer records what it can see here and nothing more.

namespace App\Service;

trait Auditing
{
    public function describe(): string
    {
        return "user service";
    }
}

class AuditLog
{
    public function record(string $message): string
    {
        return $message;
    }
}

class UserService
{
    use Auditing;

    private AuditLog $log;

    public function __construct()
    {
        $this->log = new AuditLog();
    }

    public function handle(): string
    {
        // A call to a method defined in this file.
        return $this->describe();
    }
}
